### DISCUSSION TITLE
mcserver starts with a illegal reflective access by com.thoughtworks.xstream.core.util.Fields

### DISCUSSION THREAD

**Discussion by gfuenteAr** [nextgenhealthcare/connect#4624] [Q&A]
Hi everybody, I need to resolve this situation. 

someone already did it?

WARNING: An illegal reflective access operation has occurred
WARNING: **Illegal reflective access by com.thoughtworks.xstream.core.util.Fields (file:/opt/mirthconnect/server-lib/xstream-1.4.12.jar) to field java.util.TreeMap.comparator**
WARNING: Please consider reporting this to the maintainers of com.thoughtworks.xstream.core.util.Fields
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

INFO  2021-07-06 10:30:01,873 [Main Server Thread] com.mirth.connect.server.Mirth: Mirth Connect 3.11.0 (Built on April 6, 2021) server successfully started.
INFO  2021-07-06 10:30:01,875 [Main Server Thread] com.mirth.connect.server.Mirth: This product was developed by NextGen Healthcare (https://www.nextgen.com) and its contributors (c)2005-2021.
INFO  2021-07-06 10:30:01,876 [Main Server Thread] com.mirth.connect.server.Mirth: Running OpenJDK 64-Bit Server VM 11.0.11 on Linux (5.4.17-2102.202.5.el8uek.x86_64, amd64), derby, with charset UTF-8.
INFO  2021-07-06 10:30:01,876 [Main Server Thread] com.mirth.connect.server.Mirth: Web server running at http://192.168.xxx.xxx:8080/ and https://192.168.xxx.xxx:8443/

First, I describe my server's environment:  

_operating system:_
  Virtualization: vmware
  **Operating System: Oracle Linux Server 8.4**
  5.4.17-2102.202.5.el8uek.x86_64 #2 SMP Sat May 22 16:16:03 PDT 2021 x86_64 x86_64 x86_64 GNU/Linux
  
_Java:_ 
**openjdk version "11.0.11" 2021-04-20 LTS**
OpenJDK Runtime Environment 18.9 (build 11.0.11+9-LTS)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.11+9-LTS, mixed mode, sharing)

_NextGen connect integration engine version:_
Mirth Connect **3.11.0 (Built on April 6, 2021)** 

What is the reason for the illegal reflection access error?

How can i solve it?

I installed and started the server from the root account

Thank's a lot in advance for any suggestion. 

Gustavo 
Buenos Aires - Argentina.

P.D: Sorry about my english...it isn't my native tongue.

**Comment by pacmano1** [comment]
Don't worry about it.

**Reply by gfuenteAr** [reply]
Thank's a lot!! 

Untill I learn more about it...I'll follow your recommendation!!! 

probably in next upgrades...it became a error message instead a warning. 

Have a nice week and thank you spending your time...answering me. 

Gustavo

**Comment by jonbartels** [comment]
A related issue might be https://github.com/nextgenhealthcare/connect/issues/4602 . #4602  is a fatal issue for that user but the one presented in this discussion is just a warning.

**Reply by tonygermano** [reply]
This issue is related to https://github.com/x-stream/xstream/issues/101
It looks like it has at least been partially addressed in https://github.com/x-stream/xstream/pull/216 and https://github.com/x-stream/xstream/pull/218, but I'm not sure if/when mirth is planning to upgrade xstream and pull in those changes.
